docs/project-scope.md

```diff
@@ -0,0 +1,161 @@
+| ----------------------------------------------------------- | ---------------- | ----------------------------------------------------------------------------------- |
+| “Plan a 5-day trip to Paris for a foodie couple”            | Text             | Day-by-day itinerary with food stops                                                |
+| "I want to spend a week exploring Kyiv and Lviv"            | Text             | Day-by-day itinerary with main attractions and travel route between cities included |
+| Upload a beach photo + "I want something similar in Europe" | Image + Text     | Destination guess + itinerary with beach options                                    |
+| “Make Day 2 more relaxing”                                  | Follow-up Prompt | Adjusted itinerary                                                                  |
+| Upload a blurry or unrelated photo + "I want to go here"    | Image + Text     | Generic popular destinations itinerary or last saved location itinerary             |
+| Empty or nonsensical input (e.g., “asdfghjkl”)              | Text             | Default help itinerary with sample queries                                          |
+| “Make Day 5 more exciting” when itinerary only has 3 days   | Follow-up Prompt | Revised 3-day itinerary with more activities on Day 3                               |
+
+---
+
+## 4. Core Features (MVP Milestones)
+
+- [ ] Text-to-itinerary generation with LLM
+- [ ] Prompt templates (simple & structured)
+- [ ] Basic Streamlit UI (text input + output)
+- [ ] LangChain chain for itinerary generation
+- [ ] Knowledge base (3 cities max)
+- [ ] Basic RAG with manual retrieval
+- [ ] Multimodal input (image upload)
+- [ ] Vision model integration (tags/captions)
+- [ ] Agent prototype for follow-ups or tools
+- [ ] JSON-structured output
+- [ ] Conversation History
+
+---
+
+## 5. Tech Stack
+
+- **Frontend**: Streamlit or React
+- **Backend**: FastAPI
+- **LLMs**: OpenAI GPT, Claude via AWS Bedrock
+- **LangChain / LangGraph** for orchestration
+- **Vision Models**: AWS Rekognition / CLIP
+- **Vector DB**: FAISS / Chroma
+- **Prompt Injection + RAG**
+- **Dev Tools**: Cursor IDE, Docker, GitHub Classroom
+
+---
+
+## 6. Cities Covered in Knowledge Base
+
+- [ ] Kyiv
+- [ ] Lviv
+- [ ] Add more later as part of scale-up
+
+---
+
+## 7. External APIs / Tools (Proposed)
+
+- UZ API (Ukrzaliznytsia train search)
+- OpenStreetMap (route + map rendering)
+- Tavily / SerpAPI / Eventbrite API for event search
+- Weather API (OpenWeather or Tomorrow.io)
+- Shelter location (for Ukraine-specific safety)
+- Text-to-speech (e.g., gTTS, Amazon Polly)
+- Public transport schedules (Google Maps API)
+- Travel plan transfer to calendar (Google Calendar API)
+- Reviews and working hours of places (Google Places API / Yelp API)
+- User Authentication (Auth0 / Firebase Auth)
```

Comment:
For each API, clarify if it is “planned”, “in progress”, or “explored”.
Add a note about data privacy or security for APIs handling user information.